src/compiler/tsc.ts

```diff
@@ -6,9 +6,19 @@ namespace ts {
         fileWatcher?: FileWatcher;
     }
 
-    let reportDiagnostic = reportDiagnosticSimply;
+    const defaultFormatDiagnosticsHost: FormatDiagnosticsHost = {
+        getCurrentDirectory: () => sys.getCurrentDirectory(),
+        getNewLine: () => sys.newLine,
+        getCanonicalFileName: createGetCanonicalFileName(sys.useCaseSensitiveFileNames)
+    };
+
+    let reportDiagnosticWorker = reportDiagnosticSimply;

```

Comment:
Question: why you are doing this instead of just call `reportDiagnosticSimply` directly